idxd-initcontainer: Drop libkmod, libudev

ozhuraki opened the pull request:
- Make libkmod, libudev optional
- Include accel-config, libjson-c, libuuid sources

mythi commented on build/docker/intel-idxd-initcontainer.Dockerfile:
```diff
@@ -12,38 +12,46 @@
+    apt update && apt install -y --no-install-recommends \
+        gcc make patch autoconf automake libtool pkg-config \
+        libjson-c-dev uuid-dev curl ca-certificates && \
+    cd /usr/local/src/ && apt --download-only source uuid libjson-c5 && cd /
```
Go packages are copied under `/usr/local/share/package-sources`. Let's use that here too

mythi commented on build/docker/intel-idxd-initcontainer.Dockerfile:
```diff
@@ -12,38 +12,46 @@
+    echo "$ACCEL_CONFIG_SHA256 accel-config.tar.gz" | sha256sum -c - && \
+    tar -xzf accel-config.tar.gz
+
+ADD demo/accel-config-kmod.patch /
```
time to add `demo/idxd-config` (or `accel-config`) sub-directory for all intel-idxd-initcontainer related content?

bart0sh reviewed (commented):
/lgtm

ozhuraki commented:
@mythi 
Corrected, please take a look:
- Sources are copied to /usr/local/share/package-sources/
- Moved all intel-idxd-initcontainer related content into demo/idxd-config/

mythi commented:
> Moved all intel-idxd-initcontainer related content into demo/idxd-config/

Looks like we need the Dockerfile there too and update `ci.yaml` accordingly

ozhuraki commented:
Changes:
- Renamed idxd-initcontainer to idxd-config-initcontainer as discussed
- Undone move of idxd-init.sh, dsa.conf to demo/idxd-config/ (to avoid the duplication of Dockerfile for the CI to pass)
